lib/Net/Async/Redis.pm

```diff
@@ -296,6 +306,8 @@ sub configure {
         $self->{host} //= $uri->host;
         $self->{port} //= $uri->port;
     }
+
+    die 'hashref support requires RESP3 (Redis version 6+)' if defined $self->{protocol} and $self->{protocol} eq 'resp3' and $self->{hashrefs};
```

Comment:
if I understood this correctly it should be `$self->{protocol} eq 'resp2' and $self->{hashrefs}` or the `if` should be `unless`?